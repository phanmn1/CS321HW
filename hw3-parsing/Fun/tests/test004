// Slightly larger test to produce an AST
int global = foo();

int foo(int num, boolean hah)
{
   ;
   print test(blah, blah, blah);

   return foo(num);
}
