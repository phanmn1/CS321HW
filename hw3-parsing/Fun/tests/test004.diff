1,27c1
< Globals
<   int
<   InitVarIntro
<     "global"
<     Call
<       "foo"
< Function
<   int
<   "foo"
<   Formal
<     int
<     "num"
<   Formal
<     boolean
<     "hah"
<   Block
<     Empty
<     Print
<       Call
<         "test"
<         Id("blah")
<         Id("blah")
<         Id("blah")
<     Return
<       Call
<         "foo"
<         Id("num")
---
> Invalid syntax at (17,2), (
