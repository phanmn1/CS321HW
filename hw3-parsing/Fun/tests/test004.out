Invalid syntax at (17,2), (
